pilot/pkg/xds/statusgen.go

```diff
@@ -134,11 +188,35 @@ func (sg *StatusGen) debugSyncz() []*any.Any {
 				case v3.ClusterType:
 					pxc.PerXdsConfig = &status.PerXdsConfig_ClusterConfig{}
 				}
+				additionalStatusInfo.SyncInfos[stype] = SyncInfo{
```

Comment:
can we open an issue/PR in envoy to see if we can expand CSDS to support generic types?

Reply:
Open issue for generic CSDS types - https://github.com/envoyproxy/envoy/issues/15622

Reply:
https://github.com/envoyproxy/envoy/pull/15760 - I have raised a PR. Still work in progress. The API is not yet final. As a nice side-effect, it not only allows reporting status with `proxy-status`, but also the config dumps required for `proxy-config`. In the future, it is possible to enhance `proxy-config` for new xDS types if needed.